src/components/ProviderInitializer/hooks/useSetLedgerProvider.ts

```diff
@@ -36,27 +21,59 @@ export const useSetLedgerProvider = () => {
   const logoutRoute = useSelector(logoutRouteSelector);
   const isLoggedIn = useSelector(isLoggedInSelector);
   const ledgerLogin = useSelector(ledgerLoginSelector);
+  const provider = getAccountProvider();
 
   const [ledgerData, setLedgerData] =
     useState<{
       version: string;
       dataEnabled: boolean;
     }>();
 
+  const initHWProvider = async () => {
```

Comment:
`init` and `setAddressIndex` methods could throw any exception ? If yes, I think a try-catch block would be very useful

Reply:
Resolved